src/System.Memory/src/System/Buffers/ReadOnlySequence_helpers.cs

```diff
@@ -251,19 +251,16 @@ private long GetLength(in SequencePosition start, in SequencePosition end)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private void BoundsCheck(in SequencePosition start, in SequencePosition end)
         {
-            GetTypeAndIndices(start.GetInteger(), end.GetInteger(), out SequenceType type, out int startIndex, out int endIndex);
-
+            int startIndex = GetIndex(start);
+            int endIndex = GetIndex(end);
             object startObject = start.GetObject();
             object endObject = end.GetObject();
-            if (type == SequenceType.MultiSegment && startObject != endObject)
```

Comment:
keep this type check?

Reply:
It is correct code for correct structure. Why we should check that structure is not broken in each method?

Reply:
This code does not break type safety.